source/task_scheduler.c

```diff
@@ -122,14 +127,28 @@ int aws_task_scheduler_schedule_future(struct aws_task_scheduler *scheduler,
     return AWS_OP_SUCCESS;
 }
 
-int aws_task_scheduler_run_all(struct aws_task_scheduler *scheduler) {
+int aws_task_scheduler_run_all(struct aws_task_scheduler *scheduler, uint64_t *next_task_time) {
+
+    uint64_t now;
+    if(scheduler->clock(&now)) {
+        return AWS_OP_ERR;
+    }
+
     while(1) {
         struct aws_task task_to_run = {0};
-        int err = aws_task_scheduler_next_task(scheduler, &task_to_run, 0);
-        if(AWS_OP_SUCCESS != err) {
-            return err;
+
+        if (get_next_task(scheduler, &task_to_run,  now, next_task_time)) {
+            int err_code = aws_last_error();
+            if (err_code == AWS_ERROR_TASK_SCHEDULER_NO_READY_TASKS  || err_code == AWS_ERROR_TASK_SCHEDULER_NO_TASKS) {
+                return AWS_OP_SUCCESS;
+            }
+            else if (err_code == AWS_ERROR_TASK_SCHEDULER_NO_MORE_TASKS) {
```

Comment:
It's kind of strange to report an error when you have a successful condition. Perhaps next_task_time should just be `(uint64_t)-1` in this case?